### PR TITLE
Add identity* clients to acp-client-go

### PR DESCRIPTION
**Pull request by bwereszczak**
## Description

Add identity, identityroot, identitysystem and identityself clients to acp-client-go

## Information for QA

- [ ] Is QA testing required?
- [ ] Does PR contain unit tests?
- [ ] Should QA create E2E tests for the change?

### CODE REVIEW

**Comment by bwereszczak** [comment]
@mbilski Could you check if vanity domains settings are correct for identity* clients?

**Review by mbilski — APPROVED** [review]

**Review by sburke-ce — APPROVED** [review]

File: client.go [hunk 1]
```diff
@@ -589,6 +615,42 @@ func New(cfg Config) (c Client, err error) {
...
+			c.apiPathPrefix(cfg.VanityDomainType, "/api/identity/%s"),
+			[]string{cfg.IssuerURL.Scheme},
+			client,
+		).WithOpenTracing(), nil),
+	}
+
+	c.IdentitySelf = &IdentitySelf{
+		Acp: identityself.New(httptransport.NewWithClient(
+			cfg.IssuerURL.Host,
+			c.apiPathPrefix(cfg.VanityDomainType, "/%s/%s"),
```
Comment (sburke-ce) [inline client.go:630]: I have a doubt regarding the apiPathPrefix format string.  From `acp/internal/identity/handler/rest/self/api_server.go`:
```
var APIRoutingPathPrefixes = kernel.NewCommonSubrouterPathPrefix("/{tid}/{aid}/identity")
```

File: client.go [hunk 2]
```diff
@@ -589,6 +615,42 @@ func New(cfg Config) (c Client, err error) {
...
+			[]string{cfg.IssuerURL.Scheme},
+			client,
+		).WithOpenTracing(), nil),
+	}
+
+	c.IdentityRoot = &IdentityRoot{
+		Acp: identityroot.New(httptransport.NewWithClient(
+			cfg.IssuerURL.Host,
+			c.apiPathPrefix(cfg.VanityDomainType, "/api/identity/%s"),
```
Comment (sburke-ce) [inline client.go:639]: This format string should also be reviewed. From `acp/internal/identity/handler/rest/root/api_server.go`:
```
var RoutingPathPrefixes = kernel.NewSubrouterPathPrefix(
    "/api/identity/system",
    "/api/identity/system",
    "/api/identity/system",
)
```
which would mean that there is no {tid}, even when the domain is not a vanity domain.

File: client.go (hunk 1 above)
Comment (bwereszczak) [reply client.go]: Updated

**Pull request merged**